src/DaemonExec.ts

```diff
@@ -0,0 +1,244 @@
+import fs = require('fs')
+//import path = require('path')
+import Docker = require('dockerode')
+//needs docker to already be running
+const docker = new Docker()
+
+const listImages = async () => {
+	try {
+		const images = await docker.listImages({})
+		return images
+	} catch (err) {
+		console.log(err)
+		return []
+	}
+}
+
+const listContainers = async () => {
+	try {
+		const containers = await docker.listContainers()
+		return containers
+	} catch (err) {
+		console.log(err)
+		return []
+	}
+}
+
+const importImage = async (name: string) => {
+	try {
+		const stream = await docker.pull(name)
+		return stream
+	} catch (err) {
+		console.log(err)
+	}
+}
+
+//volumePairs should be of the form [[hostSrc,containerDest]]
+const createContainer = async (
+	image: string,
+	command: string[],
+	inputDir: string,
+	stdinFile: string,
+	outputDir: string,
+	stdoutFile: string,
+	errorDir: string,
+	stderrFile: string,
+	volumePairs: [string, string][],
+) => {
+	try {
```

Comment:
The gigantic try-catch is a bad idea, we're ignoring a lot of important and unexpected errors. This function should have a way to report errors, and/or be broken up into multiple internal functions.